crates/toolchain/openvm/entry/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, ItemFn};
+
+/// Used for defining the guest's entrypoint and main function.
+///
+/// When `#![no_main]` is used, the programs entrypoint and main function is left undefined. The
```

Comment:
I was wondering, do you think it's safe to add the `#![no_main]` within the macro, where it figures out based on whether std is enabled etc? This has the benefit that on host we can make it always `std` and allows `cargo clippy` to work etc